proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -1045,13 +1046,69 @@ def initialize(blk, initialize_guess=None, state_args=None, outlvl=idaeslog.NOTS
+            var_dict["Velocity @Outlet"] = self.velocity_io[time_point, 'out']
+        var_dict['Concentration @Inlet,Membrane-Interface '] = (
+                    self.feed_side.properties_interface_in[time_point].conc_mass_phase_comp['Liq', 'NaCl'])
+        var_dict['Concentration @Outlet,Membrane-Interface '] = (
+            self.feed_side.properties_interface_out[time_point].conc_mass_phase_comp['Liq','NaCl'])
+        var_dict['Concentration @Inlet,Bulk'] = (
+                    self.feed_side.properties_in[time_point].conc_mass_phase_comp['Liq', 'NaCl'])
+        var_dict['Concentration @Outlet,Bulk'] = (
+            self.feed_side.properties_out[time_point].conc_mass_phase_comp['Liq','NaCl'])
+        if self.feed_side.properties_interface_out[time_point].is_property_constructed('pressure_osm'):
+            var_dict['Osmotic Pressure @Outlet,Membrane-Interface '] = (
+                self.feed_side.properties_interface_out[time_point].pressure_osm)
+        if self.feed_side.properties_out[time_point].is_property_constructed('pressure_osm'):
+            var_dict['Osmotic Pressure @Outlet,Bulk'] = (
+                self.feed_side.properties_out[time_point].pressure_osm)
+        if self.feed_side.properties_interface_in[time_point].is_property_constructed('pressure_osm'):
+            var_dict['Osmotic Pressure @Inlet,Membrane-Interface'] = (
+                self.feed_side.properties_interface_in[time_point].pressure_osm)
+        if self.feed_side.properties_in[time_point].is_property_constructed('pressure_osm'):
+            var_dict['Osmotic Pressure @Inlet,Bulk'] = (
+                self.feed_side.properties_in[time_point].pressure_osm)
+        if self.feed_side.properties_in[time_point].is_property_constructed('flow_vol_phase'):
+            var_dict['Volumetric Flowrate @Inlet'] = (
+                self.feed_side.properties_in[time_point].flow_vol_phase['Liq'])
+        if self.feed_side.properties_out[time_point].is_property_constructed('flow_vol_phase'):
+            var_dict['Volumetric Flowrate @Outlet'] = (
+                self.feed_side.properties_in[time_point].flow_vol_phase['Liq'])
+
+        # TODO: (1) add more vars, (2) would be nice to add units to output, and (3) should be able to report output of
+        #  "NaN" or "Not Reported", mainly for properties that exist but are not necessarily constructed within model
+        #  constraints. One example in this case is the osmotic pressure of the bulk feed, which would certainly be of
+        #  interest to users with a background in desalination (it is currently not reported because it is not directly
+        #  used in any model constraints). Furthermore, the report() method seems to be limited to Pyomo Var objects for
+        #  which the pyomo value() method is applied to. That is, a Pyomo Var object must be used; e.g., providing a
+        #  list as output would yield an error.
```

Comment:
It seems like you can provide Expressions as `"exprs" : exprs_dict` and Params as `"params" : params_dict` in the return dictionary below:
https://github.com/IDAES/idaes-pse/blob/b878aea16c073e813779283d8aa997de429fbe75/idaes/core/process_base.py#L226-L323

Expressions might be useful for some of these derived properties here and elsewhere.

Reply:
Nice - thanks for checking this. It also looks like we'd want to modify one of the args in `tabular_writer()` to add another column that displays units; this is not a necessity but I think it would be useful and would provide more transparency for the user's sake.